java/casting.md

```diff
@@ -9,3 +9,17 @@ Mit einem cast lässt sich nicht jedes Objekt X in ein Objekt Y umwandeln. Es ka
 Beispielsweise kann man einen `CommandSender` zu `Player` casten, weil `Player` von `CommandSender` erbt. Es gibt aber noch andere `CommandSender` - z.B. die Konsole oder Command Blocks - deshalb sollte man vorher mit `instanceof` überprüfen, ob es sich wirklich um einen `Player` handelt.
 
 Man kann jedoch nicht etwa ein `BlockBreakEvent` zu einem `Player` casten, auch wenn ein `Player` etwas mit dem event zu tun hat. `BlockBreakEvent` und `Player` sind **nicht verwandt**, somit wird der cast **immer** einen Fehler erzeugen. In solchen Fällen sollte man schauen, ob es einen Weg gibt, um an das Objekt zu kommen, was man haben will. Beim `BlockBreakEvent` z.B. gibt es eine Methode `getPlayer()`, mit der man den `Player` erhält.
+
+Auf der nächsten `Seite` findest du einen einfachen Syntax dafür.
+<new_page>
+Das in Java 14 eingeführte `Pattern Matching` ist dabei sehr nützlich - das führt zu simplerem und einfacherem Code.
+```java
+if (sender instanceof Player player) {
+  player.sendMessage("hello")
+}
+```
+Ohne eine neue Variable zu definieren, ist es möglich, `player` zu verwenden. `player` ist hierbei eine lokale Variable vom Typ `Player`. Alternativ kannst du auch negieren, mit einem Early Return Pattern.
```

Comment:
Technically, you still define a new variable holding the same value.

Reply:
Should it be "Ohne eine explizite Variable mit Typ zu definieren und explizit einen Cast auszuführen"?

Reply:
`Player player` seems like an explicit variable declaration to me though.

Reply:
Wir können jetzt natürlich über explizit und implizit streiten. Leute die hiermit Probleme haben werden mit unseren drölf zusatzinformationen aber erstmal nichts anfangen können. Es added also sehr wenig value und verwirrt die leute vermutlich eher als ihnen zu helfen. Leute müssen wissen, dass sie die variable nicht vor dem cast deklarieren müssen und das tun sie.